### ISSUE TITLE
RFC: power off screen on STOP

### ISSUE THREAD

**Issue by christf** [sosedoff/omxremote#12]
When finishing a media file, it could be of benefit to turn off the screen of the raspberrypi - and to re-enable the screen when starting playback.

This can be achieved with:
```
vcgencmd display_power 0
vcgencmd display_power 1
```
First this could be hard-coded or even enabled via cli options by specifying --run-command-on-play "vcgencmd display_power 1" --run-command-on-stop "vcgencmd display_power 0"

What are your thoughts?

**Comment by sosedoff** [comment]
I've not used this command before, does it always get shipped with Raspbian? If yes - sure, we can add something like that (`-exec-before`, `-exec-after` flags).

**Comment by christf** [comment]
it seems to be shipped with raspbian: https://www.raspberrypi.org/documentation/raspbian/applications/vcgencmd.md I had not used it before either. allowing to trigger external commands could be interesting to power off devices that are connected via FS20 gear too. In that case a shell script would be executed that powers off the projector after 10 minutes...

**Comment by pietrondo** [comment]
@christf  but when the video finish it start the "on-stop command"? thx

**Comment by christf** [comment]
yes it does and I feel this is the intended behavior. What are indications against it?

**Comment by pietrondo** [comment]
It doesn't work... after 10 sec the screen reapper with console log.

**Comment by pietrondo** [comment]
/usr/bin/omxremote -run-on-play "vcgencmd display_power 1" -run-on-stop "vcgencmd display_power 0" /home/pi/media

**Comment by pietrondo** [comment]
I have tried also /usr/bin/omxremote1 -media /home/pi/media -run-on-play vcgencmd display_power 1 -run-on-stop vcgencmd display_power 0 && vcgencmd display_power 0 .
But after the file is played.. the screen reapper with console log. i only want that that vcgencmd display_power 0 is executed :-)
thx

**Comment by christf** [comment]
On Tue, Jul 14, 2020 at 01:47:09AM -0700, Pietro wrote:
>I have tried also /usr/bin/omxremote1 -media /home/pi/media -run-on-play vcgencmd display_power 1 -run-on-stop vcgencmd display_power 0 && vcgencmd display_power 0 .
>But after the file is played.. the screen reapper with console log. i only want that that vcgencmd display_power 0 is executed :-)
>thx

I do not think the current implementation supports parameters.
Does it work for you if you place your command in a shell script and run 
it from the script?


-- 
()  ascii ribbon campaign - against html e-mail
/\  www.asciiribbon.org   - against proprietary attachments

**Comment by pietrondo** [comment]
yes... i found now the solution. I made two shell script. thx ;-)

**Comment by pietrondo** [comment]
excuse i have another request. during the run-on-play script, it could be possibile that i recall the omxremote... so at the end have two istances of omxplayer (my on-play script take some minutes).
Is possible to "prevent" another omxremote call during "run on play script"? how can i do?
thx

**Comment by christf** [comment]
the solution to this is to place a lock file as part of your script and then make a choice to terminate one of the two instances
